src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -758,6 +758,11 @@ private CorInfoHelpFunc getCastingHelper(ref CORINFO_RESOLVED_TOKEN pResolvedTok
         private CorInfoHelpFunc getNewHelper(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_METHOD_STRUCT_* callerHandle, byte* pHasSideEffects = null)
         {
             TypeDesc type = HandleToObject(pResolvedToken.hClass);
+            MetadataType metadataType = type as MetadataType;
+            if (metadataType != null && metadataType.IsAbstract)
+            {
+                throw new RequiresRuntimeJitException(metadataType);
```

Comment:
I would throw some kind of `TypeSystemException` here instead. `RequiresRuntimeJitException` is not a great fit IMO because we don't be able to new up the abstract type at runtime either.  Crossgen throws an invalid operation exception in this case. How about `InvalidProgramException`?